src/task/syncBorrowedShips.ts

```diff
@@ -0,0 +1,190 @@
+import moment from 'moment';
+import { getAccessToken } from '../data-source/accessToken/accessToken';
+import { ESI_UNIVERSE_STRUCTURES_$structureId } from '../data-source/esi/endpoints';
+import { fetchEsi } from '../data-source/esi/fetch/fetchEsi';
+import { fetchEveNames } from '../data-source/esi/names';
+import { dao } from '../db/dao';
+import { CharacterShipRow } from '../db/dao/CharacterShipDao';
+import { Tnex } from '../db/tnex';
+import { Asset, fetchAssets, formatLocationFlag } from '../eve/assets';
+import { TYPE_CATEGORY_SHIP } from '../eve/constants/categories';
+import { buildLoggerFromFilename } from '../infra/logging/buildLogger';
+import { JobLogger } from '../infra/taskrunner/Job';
+import { Task } from '../infra/taskrunner/Task';
+import { arrayToMap } from '../util/collections';
+
+const logger = buildLoggerFromFilename(__filename);
+
+export const syncBorrowedShips: Task = {
+  name: 'syncBorrowedShips',
+  displayName: 'Sync borrowed ships',
+  description: "Searches for corp-owned ships in members' assets.",
+  timeout: moment.duration(60, 'minutes').asMilliseconds(),
+  executor,
+};
+
+/**
+ * Caches ID-to-name mapping for NPC stations and player structures. Station
+ * IDs are already cached globally, but we are copying them to this cache for
+ * consitency of access. Player structure names are mutable, so we only keep
+ * this cache for the duration of a single sync.
+ */
+class LocationCache {
+  private cache = new Map<number, string>();
+
+  async cachePlayerStructure(sid: number, token: string) {
+    const name = this.cache.get(sid);
+    if (name !== undefined) return;
+    const structureData = await fetchEsi(ESI_UNIVERSE_STRUCTURES_$structureId, {
+      structureId: sid,
+      _token: token,
+    });
+    this.cache.set(sid, structureData.name);
+  }
+
+  async cacheStations(ids: number[]) {
+    const station_names = await fetchEveNames(ids);
+    for (let x of ids) {
+      this.cache.set(x, station_names[x]);
+    }
+  }
+
+  get(x: number) {
+    return this.cache.get(x);
+  }
+}
+
+const CORP_OWNED_SHIP_NAME_PREFIX = 'SA ';
```

Comment:
Could you shove this definition up to the top of the module so the high-level configuration is more obvious?